### PR TITLE
ENH: V72 release changes

### PR DESCRIPTION
**Pull request by mbechtold**
This PR includes all scientific code changes related to the V72 release. See related release notes published at the FAO AquaCrop website.
The PR also includes 
-cleanup of unused variables in fortran code
-two corrections of testing floating point numbers on equality (now making appropriate use of epsilon)
-a bug fix (difference with old Pascal code) in the subroutine CalculateEffectiveRainfall which did not affect results of the testcases. 

Test results (DEBUG=0 and DEBUG=1):
CassavaTest.PRO
Alf2Isp50.PRM
Ottawa.PRM
CassavaTestMonth.PRO (with monthly meteo data as input)
--> zero-diff with pascal executable apart from rounding differences (last digit) due to rounding digits / insignificant digits (extra digits that extend beyond the number of digits allowed by the given precision), like e.g. 0.05000000030935 or 0.04999999956757 are rounded to either 0.1 or 0.0 in fortran and pascal output.

### CODE REVIEW

File: src/run.f90 [hunk 1]
```diff
@@ -572,68 +596,13 @@ module ac_run
...
                          ! irrigation water was not yet recorded
-
+! Version 7.2
+integer(int32) :: LastIrriDAP
 
 
 contains
 
 
-subroutine open_file(fhandle, filename, mode, iostat)
```
Comment (lbusschaert) [inline src/run.f90:581]: Why are those removed? not used?

File: src/run.f90 [hunk 2]
```diff
@@ -2182,136 +2211,6 @@ subroutine SetTransfer_Bmobilized(Bmobilized)
 end subroutine SetTransfer_Bmobilized
 
 
-!TminDatSet
```
Comment (lbusschaert) [inline src/run.f90:2185]: Again, are those unused? Or moved?

Comment (lbusschaert) [reply src/run.f90:2185]: Ok moved to tempprocessing

File: src/utils.f90 [hunk 3]
```diff
@@ -248,4 +248,61 @@ subroutine threestrings2threepointers(string1, string2, string3, &
     c_pointer3 = c_loc(f_string3)
 end subroutine threestrings2threepointers
 
+
+subroutine write_file(fhandle, line, advance, iostat)
```
Comment (lbusschaert) [inline src/utils.f90:252]: Ok, I see they are moved

**Review by lbusschaert — CHANGES_REQUESTED** [review]
Looks great, just some minor (stylistic comments).
I would still wait until Dirk confirms that the fortran standalone is working before we merge it

File: src/run.f90 (hunk 1 above)
Comment (mbechtold) [reply src/run.f90:581]: moved due to solving circularity in the compilation

File: src/utils.f90 (hunk 3 above)
Comment (mbechtold) [reply src/utils.f90:252]: yes

**Comment by mbechtold** [comment]
Thanks, your suggestions are integrated!

File: src/run.f90 [hunk 4]
```diff
@@ -7051,7 +7022,7 @@ subroutine AdvanceOneTimeStep(WPi, HarvestNow)
         call DetermineRootZoneWC(GetRootingDepth(), SWCtopSoilConsidered_temp)
         call SetSimulation_SWCtopSoilConsidered(SWCtopSoilConsidered_temp)
         ! temperature stress affecting crop transpiration
-        if (GetCCiActual() <= 0.0000001_dp) then
+        if (GetCCiActual() <= 0.000001_dp) then
```
Comment (gdelannoy) [inline src/run.f90:7025]: replace 0.0000001_dp with a preset constant?

File: src/run.f90 [hunk 5]
```diff
@@ -7534,7 +7508,7 @@ subroutine WriteDailyResults(DAP, WPi)
         end if
 
         ! 4. Air temperature stress
-        if (GetCCiActual() <= 0.0000001_dp) then
+        if (GetCCiActual() <= 0.000001_dp) then
```
Comment (gdelannoy) [inline src/run.f90:7511]: idem

File: src/run.f90 [hunk 6]
```diff
@@ -7547,7 +7521,7 @@ subroutine WriteDailyResults(DAP, WPi)
         end if
 
         ! 5. Relative cover of weeds
-        if (GetCCiActual() <= 0.0000001_dp) then
+        if (GetCCiActual() <= 0.000001_dp) then
```
Comment (gdelannoy) [inline src/run.f90:7524]: idem, and perhaps we'd need a full code search on these types of hardwired thresholds

File: src/simul.f90 [hunk 7]
```diff
@@ -3401,7 +3401,7 @@ subroutine DetermineCCiGDD(CCxTotal, CCoTotal, &
             else
                 if (GetCCiPrev() < (0.97999_dp*CCxSF)) then
                     call DetermineGDDCGCadjusted(GDDCGCadjusted)
-                    if (GDDCGCadjusted > 0.00000001_dp) then
+                    if (GDDCGCadjusted > 0.000001_dp) then
```
Comment (gdelannoy) [inline src/simul.f90:3404]: idem

File: src/simul.f90 [hunk 8]
```diff
@@ -4142,7 +4146,7 @@ subroutine PrepareStage1()
 
     Soil_temp = GetSoil()
 
-    if (GetSurfaceStorage() > 0.0000001_dp) then
+    if (GetSurfaceStorage() > 0.000001_dp) then
```
Comment (gdelannoy) [inline src/simul.f90:4149]: idem

File: src/simul.f90 [hunk 9]
```diff
@@ -4423,7 +4427,7 @@ subroutine ExtractWaterFromEvapLayer(EvapToLose, Zact, Stg1)
         end if
         Ztot = Ztot + fracZ * (GetCompartment_Thickness(compi))
         if ((Compi >= GetNrCompartments()) &
-            .or. (abs(StillToExtract) < 0.0000001_dp) &
+            .or. (abs(StillToExtract) < 0.000001_dp) &
```
Comment (gdelannoy) [inline src/simul.f90:4430]: idem

File: src/simul.f90 [hunk 10]
```diff
@@ -4423,7 +4427,7 @@ subroutine ExtractWaterFromEvapLayer(EvapToLose, Zact, Stg1)
...
             .or. (Ztot >= 0.999999_dp*Zact)) exit loop
```
Comment (gdelannoy) [inline src/simul.f90:4431]: idem, but a different constant

File: src/simul.f90 [hunk 11]
```diff
@@ -4770,7 +4774,7 @@ subroutine DetermineCCi(CCxTotal, CCoTotal, StressLeaf, FracAssim, &
             else
                 if (GetCCiPrev() < 0.97999_dp*CCxSF) then
```
Comment (gdelannoy) [inline src/simul.f90:4775]: arbitrarily set to 0.97... or with a purpose?

File: src/simul.f90 [hunk 12]
```diff
@@ -4770,7 +4774,7 @@ subroutine DetermineCCi(CCxTotal, CCoTotal, StressLeaf, FracAssim, &
...
                     call DetermineCGCadjusted(CGCadjusted)
-                    if (CGCadjusted > 0.00000001_dp) then
+                    if (CGCadjusted > 0.000001_dp) then
```
Comment (gdelannoy) [inline src/simul.f90:4777]: idem

File: src/simul.f90 [hunk 13]
```diff
@@ -5639,12 +5642,12 @@ subroutine BUDGET_module(dayi, TargetTimeVal, TargetDepthVal, VirtualTimeCC, &
             call CalculateEvaporationSurfaceWater
         end if
         ! stage 1 evaporation
-        if ((abs(GetEpot() - GetEact()) > 0.0000001_dp) &
+        if ((abs(GetEpot() - GetEact()) > 0.000001_dp) &
```
Comment (gdelannoy) [inline src/simul.f90:5645]: idem

File: src/simul.f90 [hunk 14]
```diff
@@ -5639,12 +5642,12 @@ subroutine BUDGET_module(dayi, TargetTimeVal, TargetDepthVal, VirtualTimeCC, &
...
             .and. (GetSimulation_EvapWCsurf() > 0._dp)) then
             call CalculateSoilEvaporationStage1()
         end if
         ! stage 2 evaporation
-        if (abs(GetEpot() - GetEact()) > 0.0000001_dp) then
+        if (abs(GetEpot() - GetEact()) > 0.000001_dp) then
```
Comment (gdelannoy) [inline src/simul.f90:5650]: idem

File: src/utils.f90 [hunk 15]
```diff
@@ -64,15 +64,15 @@ function GetReleaseDate() result(str)
     !! Returns a string containing the month and year of the release.
     character(len=:), allocatable :: str
 
-    str = 'August 2023'
+    str = 'August 2024'
```
Comment (gdelannoy) [inline src/utils.f90:67]: Not sure if this matters at all, but it would be September/October in reality

**Review by gdelannoy — COMMENTED** [review]
Impressive work! I only have minor suggestions, but the code looks technically ready otherwise.

File: src/global.f90 [hunk 16]
```diff
@@ -7953,7 +7981,7 @@ subroutine CalculateETpot(DAP, L0, L12, L123, LHarvest, DayLastCut, CCi, &
         end if
 
         ! Correction for Air temperature stress
-        if ((CCiAdjusted <= 0.0000001_dp) &
+        if ((CCiAdjusted <= 0.000001_dp) &
```
[the remark this answers is not in the document]
Comment (mbechtold) [reply src/global.f90]: Good point. Replaced this new single precision threshold that is used in AquaCrop with a constant called ac_zero_threshold. All 0.000001_dp were replaced.

File: src/run.f90 (hunk 4 above)
Comment (mbechtold) [reply src/run.f90]: done

File: src/run.f90 (hunk 5 above)
Comment (mbechtold) [reply src/run.f90]: done

File: src/run.f90 (hunk 6 above)
Comment (mbechtold) [reply src/run.f90]: done

File: src/simul.f90 (hunk 7 above)
Comment (mbechtold) [reply src/simul.f90]: done

**Comment by mbechtold** [comment]
Thanks for the review. Changes were made.

**Review by lbusschaert — APPROVED** [review]

**Review by gdelannoy — APPROVED** [review]
